### PR TITLE
feat: Support Max Parameters and Properties

### PR DESCRIPTION
**Pull request by BrandonStalnaker**
## Summary
 - GA places limitations on the number of parameters and properties allowed on their events and items.

 ## Testing Plan
 - Tested manually and with unit tests

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5278

### CODE REVIEW

**Comment by BrandonStalnaker** [comment]
Seems like the tests are only failing on development so I'm not sure what's going on. Any idea Mark?

**Review by einsteinx2 — CHANGES_REQUESTED** [review]
Looks like some existing unit tests failed with this change.

File: src/main/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4Kit.kt [hunk 1]
```diff
@@ -114,6 +123,13 @@ class GoogleAnalyticsFirebaseGA4Kit : KitIntegration(), KitIntegration.EventList
         commerceEvent.promotions?.let {
             for (promotion in it) {
                 bundle = getPromotionCommerceEventBundle(promotion).bundle
+                if (bundle.keySet().size > eventMaxParameterProperty) {
```
[the remark this answers is not in the document]
Comment (BrandonStalnaker) [reply src/main/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4Kit.kt]: Extracted the code to its own function. Any idea what's going on around the tests? The java.lang.NoSuchMethodError doesn't make any sense to me sense it compiles successfully. Seems like a test specific issue?

[the remark this answers is not in the document]
Comment (markvdouw) [reply src/main/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4Kit.kt]: So strange it doesnt make sense!

**Review by markvdouw — APPROVED** [review]
LGTM

**Pull request merged**